### PR TITLE
Added sceAppMgrAppParamGetString implementation.

### PR DESCRIPTION
**Pull request by Rinnegatamante**
Untested since i was unable to find a proper homebrew to test with.

### CODE REVIEW

File: src/emulator/host/include/host/state.h [hunk 1]
```diff
@@ -17,6 +17,7 @@
 
 #pragma once
 
+#include "../../sfo.h"
```
Comment (VelocityRa) [inline src/emulator/host/include/host/state.h]: don't use relative include paths

File: src/emulator/vpk.cpp [hunk 2]
```diff
@@ -45,11 +45,11 @@ static size_t write_to_buffer(void *pOpaque, mz_uint64 file_ofs, const void *pBu
     return n;
 }
 
-bool load_vpk(Ptr<const void> &entry_point, std::string& game_title, std::string& title_id, IOState &io, MemState &mem, const std::wstring& path) {
+bool load_vpk(Ptr<const void> &entry_point, std::string &game_title, std::string &title_id, IOState &io, MemState &mem, const std::wstring &path, SfoFile &sfo_file) {
```
[the remark this answers is not in the document]
Comment (Rinnegatamante) [reply src/emulator/vpk.cpp]: Aren't those all output parameters?

File: src/emulator/modules/SceAppMgr/SceAppMgrUser.cpp [hunk 3]
```diff
@@ -53,8 +54,14 @@ EXPORT(int, sceAppMgrAppParamGetInt) {
     return unimplemented("sceAppMgrAppParamGetInt");
 }
 
-EXPORT(int, sceAppMgrAppParamGetString) {
-    return unimplemented("sceAppMgrAppParamGetString");
+EXPORT(int, sceAppMgrAppParamGetString, int pid, int param, char *string, int length) {
+    std::string res = get_data(host.sfo_handle, param);
+#ifdef WIN32
```
Comment (VelocityRa) [inline src/emulator/modules/SceAppMgr/SceAppMgrUser.cpp]:
```cpp
res.copy(string, length)
```

File: src/emulator/vpk.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (bentokun) [reply src/emulator/vpk.cpp]: The path is not, it's just a reference to the string, not write anything.

File: src/emulator/host/include/host/state.h (hunk 1 above)
Comment (Rinnegatamante) [reply src/emulator/host/include/host/state.h]: <sfo.h> doesn't work.

Comment (VelocityRa) [reply src/emulator/host/include/host/state.h]: we only use absolute paths: `<host/sfo.h>`

Comment (Rinnegatamante) [reply src/emulator/host/include/host/state.h]: sfo's not in host directory.

Comment (VelocityRa) [reply src/emulator/host/include/host/state.h]: yea nvm, have you tried adding emulator to CMakeLists.txt?

File: src/emulator/vpk.h [hunk 4]
```diff
@@ -17,11 +17,12 @@
 
 #pragma once
 
+#include "sfo.h"
```
Comment (petmac) [inline src/emulator/vpk.h]: We should forward-declare SfoFile rather than including sfo.h.

File: src/emulator/host/include/host/state.h (hunk 1 above)
Comment (petmac) [reply src/emulator/host/include/host/state.h]: Looks like SFO stuff needs to be moved to a new library.

File: src/emulator/modules/SceAppMgr/SceAppMgrUser.cpp [hunk 5]
```diff
@@ -53,8 +54,10 @@ EXPORT(int, sceAppMgrAppParamGetInt) {
     return unimplemented("sceAppMgrAppParamGetInt");
 }
 
-EXPORT(int, sceAppMgrAppParamGetString) {
-    return unimplemented("sceAppMgrAppParamGetString");
+EXPORT(int, sceAppMgrAppParamGetString, int pid, int param, char *string, int length) {
+    std::string res = get_data(host.sfo_handle, param);
+	res.copy(string, length);
```
Comment (VelocityRa) [inline src/emulator/modules/SceAppMgr/SceAppMgrUser.cpp]: nit: tab -> space

**Comment by Rinnegatamante** [comment]
Already tried twice to solve conflicts unsucessfully. If someone wants to try, feel free to do so. If not, just close this PR.

**Pull request merged**